packages/hardhat-verify/src/internal/errors.ts

```diff
@@ -421,25 +422,23 @@ Encoder error reason: ${reason} fault in ${operation}`,
   }
 }
 
+/**
+ * `VerificationAPIUnexpectedMessageError` is thrown when the block explorer API
+ * does not behave as expected, such as when it returns an unexpected response message.
+ */
 export class VerificationAPIUnexpectedMessageError extends HardhatVerifyError {
   constructor(message: string) {
     super(`The API responded with an unexpected message.
-Please report this issue to the Hardhat team.
 Contract verification may have succeeded and should be checked manually.
 Message: ${message}`);
   }
 }
 
-export class UnexpectedError extends HardhatVerifyError {
-  constructor(e: unknown, functionName: string) {
-    const defaultErrorDetails = `Unexpected error in ${functionName}`;
-    const errorDetails =
-      e instanceof Error
-        ? e.message ?? defaultErrorDetails
-        : defaultErrorDetails;
-    super(`An unexpected error occurred during the verification process.
-Please report this issue to the Hardhat team.
-Error Details: ${errorDetails}`);
+export class NetworkRequestError extends HardhatVerifyError {
+  constructor(e: Error) {
+    super(
+      `A network request failed. This error is not related to Hardhat. Original error: ${e.message}`
```

Comment:
```suggestion
      `A network request failed. This is an error from the block explorer, not Hardhat. Error: ${e.message}`
```
Is block explorer the right term here? I think `Original error` makes more sense from our point of view, but less so from a user's perspective.